code/__HELPERS/icon_smoothing.dm

```diff
@@ -226,25 +227,30 @@
 		else if(adjacencies & N_EAST)
 			se = "4-e"
 
-	if(A.top_left_corner != nw)
-		A.overlays -= A.top_left_corner
-		A.top_left_corner = nw
-		A.add_overlay(nw)
+	var/list/smooth_overlays = A.overlay_list[A.smooth_overlay_channel]["overlay_smooth"]
+	if(!smooth_overlays)
+		smooth_overlays = list()
+		A.overlay_list[A.smooth_overlay_channel]["overlay_smooth"] = smooth_overlays
 
-	if(A.top_right_corner != ne)
-		A.overlays -= A.top_right_corner
-		A.top_right_corner = ne
-		A.add_overlay(ne)
+	var/changed = FALSE
+	if(smooth_overlays["top_left_corner"] != nw)
```

Comment:
Searching for a list key is faster then searching for a datum var in all but certain extreme cases.